text/0000-input-macros.md

```diff
@@ -0,0 +1,484 @@
+cases trim newlines by default. In Go, the `bufio.Reader.ReadString()` function
+does not trim newlines, but the `bufio.Scanner` type does. The `bufio.Scanner`
+type is the one that is used in the Go standard library for reading input
+from the console. The `bufio.Scanner` type is a higher level abstraction.
+
+## Why should the function handle EOF?
+
+If the function performs newline trimming, it also has to return an error when
+the input stream reaches EOF because otherwise users had no chance of detecting
+EOF. Handling EOF allows for example a program that attempts to parse each line
+as a number and prints their sum on EOF to be implemented as follows:
+
+```rs
+fn main() -> Result<(), InputError<ParseFloatError>> {
+    let mut sum = 0;
+    loop {
+        let result: Result<f64, _> = input!();
+        match result {
+            Ok(mut prices) => sum += prices,
+            Err(InputError::Eof) => {
+                println!("Oh no, EOF!");
+                break;
+            }
+            Err(e) => {
+                println!("Error: {:?}", e); // e could be a ParseFloatError
+                                            // or an IO error
+                return Err(e);
+            }
+        }
+    }
+    println!("{}", sum);
+    Ok(())
+}
+```
+
+## What is the impact of not doing this?
+If this RFC is not accepted, the current way of reading input from the user
+will remain the same. This means that new Rustaceans will have to learn how to
+use the `std::io::Stdin::read_line` function and how to handle the `Buffer`
+and the `String` types. This can be a barrier to entry for new Rustaceans and
+it can make the language less friendly. 
+
+This RFC was presented as a pre-RFC at a [Rust Argentina meetup](https://rust-lang.ar/p/2025-april/), 
+where it received positive feedback, particularly from attendees new to Rust 
+(many with backgrounds in NodeJS, Python, and Go). They found the current 
+approach to reading input in Rust complex and not very user-friendly, and were 
+enthusiastic about the proposed macros.
+
+They were not too happy with the current way of reading input from the user. 
+They think that it was too complex and not too friendly. 
+
+The presentation can be found [here](https://youtu.be/CjZq93pzOkA?t=4080) (in 
+Spanish)
+And yes, the guy with the black shirt with the Rust logo is me (👋).
+
+## Could this be done in a library or macro instead? 
+
+Well yes, but I think that this is a good idea to have it in the standard
+library. I think that this is a good idea to have it in the standard library
+because it is a common use case and it is a good idea to have it in the
+standard library to make the language more friendly.
+
+Another way that we could consider this is added like a pseudo oficial library 
+like `rand` which is not in the standard library but is recommended by the Rust 
+team and the oficial documentation as the book does it.
+
+# Prior art
```

Comment:
Remark: AFAICT this RFC in its current form is slightly light on how it evolved to the *current* design -- you backlinked a lot of discussions (which are of course very helpful), but I'm not sure if alternative proposals were discussed in terms of their trade-offs, and how it's determined that *this* design has the best tradeoffs in your opinion.

Reply:
Thanks for noticing that. Indeed, there were several alternatives discussed in the linked issues, but you're right we could do a better job summarizing how we arrived at the current design and why we consider it the most balanced.  
To summarize: this is currently the simplest design, and one of the most frequently proposed in the discussions (this has been suggested since 2014, lol). I think if everyone keeps arriving at the same design (even without reading all the discussions), it’s probably because it’s the most intuitive one.

Something similar happened in Python, where the function name `input` was initially rejected and other alternatives were considered, but they eventually went with `input`.